locals.management_groups.tf

```
@@ -360,3 +361,14 @@ locals {
     if contains(keys(azurerm_management_group.level_5), try(length(value.parent_management_group_id) > 0, false) ? "${local.provider_path.management_groups}${value.parent_management_group_id}" : local.empty_string)
   }
 }
+
+# The following local is used to build the list of Management Groups 
+# that will have Diagnostic Settings deployed, based on boolean varaible 
+# deploy_diagnostics_for_mg
+locals {
+  azapi_diagnostics = [
+    for mg_id in keys(local.es_landing_zones_map) :
+    mg_id
+    if local.deploy_diagnostics_for_mg
+  ]
+}
```

```suggestion
locals {
  azapi_mg_diagnostics = {
    for mg_id in keys(local.es_landing_zones_map) :
    mg_id => ""
    if local.deploy_diagnostics_for_mg
  }
}
```

Hi @lachaves please can you make this a map and add the mg term to the name of the local. a bit like the suggestion?